src/t8_element_cxx.hxx

```diff
@@ -741,21 +691,28 @@ struct t8_eclass_scheme
    * \param [in] length   The number of elements to be initialized.
    * \param [in,out] elems On input an array of \b length many allocated
    *                       elements.
-   * \param [in] called_new True if the elements in \a elem were created by a call
-   *                       to \ref t8_element_new. False if no element in \a elem
-   *                       was created in this way. The case that only some elements
-   *                       were created by \ref t8_element_new should never occur.
    * \note In debugging mode, an element that was passed to \ref t8_element_init
    * must pass \ref t8_element_is_valid.
    * \note If an element was created by \ref t8_element_new then \ref t8_element_init
    * may not be called for it. Thus, \ref t8_element_new should initialize an element
    * in the same way as a call to \ref t8_element_init would.
-   * Thus, if \a called_new is true this function should usually do nothing.
    * \see t8_element_new
    * \see t8_element_is_valid
    */
   virtual void
-  t8_element_init (int length, t8_element_t *elem, int called_new) const
+  t8_element_init (int length, t8_element_t *elem) const
+    = 0;
+
+  /** Deinitialize an array of allocated elements.
+   * \param [in] length   The number of elements to be deinitialized.
+   * \param [in,out] elems On input an array of \b length many allocated
+   *                       and initialized elements, on output an array of
+   *                       \b length many allocated, but not initialized elements.
+   * \note Call this function if you called t8_element_init on the element pointers.
+   * \see t8_element_init
+   */
+  virtual void
+  t8_element_deinit (int length, t8_element_t *elem) const
```

Comment:
It does nothing for the current scheme, but the whole point of the `init` function is to allow dynamic allocation inside an element, that was not created by new, but by allocating `elem_size * count` many bytes. So this memory inside the element needs to be freed as well, but we cannot call `t8_element_destroy`, as it would also free the element container memory, but we are responsible for that ourselves.
Thus, new has the counterpart destroy, and init has the counterpart deinit.
For current schemes, init/deinit does nothing, but I wanted to remove the init function for the same reason as you wanted to remove the deinit function, and was told that we need to support dynamic allocation in elements.
In my opinion, we do not really support that, unless we also have a deinit function.

Reply:
I also tried to improve the documentation on the init and deinit functionality